### ISSUE TITLE
Support KVStore array type

### ISSUE THREAD

**Issue by rtoelhoej** [LukeMurphey/lookup-editor#41]
We are using the Lookup editor app to view KVStore values and modify/import values.

The Lookup editor app doesn't support KVStore field array type, if you for example check/uncheck a checkbox via the lookup editor, then the array field value will be converted to an string, causing failures in our searches and python scripts.

https://docs.splunk.com/Documentation/Splunk/7.0.2/RESTREF/RESTkvstore defines the types that can be used for KVStores. (array, number, boolean, time, string, cidr)

**Generate collection and lookup:**

Add to default\collections.conf:

```
[test_array]
field.stringValue = string
field.numberValue = number
field.dateValue = time
field.arrayValue = array
field.booleanValue = boolean
field.cidrValue = cidr
```

Add to default\transforms.conf

```
[test_array_lookup]
external_type = kvstore
collection = test_array
fields_list = _key,stringValue,numberValue,dateValue,arrayValue,booleanValue,cidrValue
```

**Generate test data:**

```
| makeresults | eval stringValue = "String Value" | eval numberValue = 4432.321 | eval arrayValue = split("Test1;Test2;Test3;Test4",";") | eval dateValue = now() | eval booleanValue = 1 | eval cidrValue = "127.0.0.1" | outputlookup test_array_lookup append=true
```

**Search data**
Array value is split per line and is a true array.

`| inputlookup test_array_lookup | eval key = _key`

![image](https://user-images.githubusercontent.com/8926536/56946362-2e3c1a00-6b2a-11e9-9e57-dc26ab6bb16d.png)

**Edit non array field in lookup editor**
For example click on booleanValue to enable it.

![image](https://user-images.githubusercontent.com/8926536/56946405-4f046f80-6b2a-11e9-8075-d7a486ee5932.png)

**Perform search again, array value is converted to normal string**

![image](https://user-images.githubusercontent.com/8926536/56946461-6c393e00-6b2a-11e9-9119-8c7144ce4f8c.png)

**Comment by LukeMurphey** [comment]
Thanks for the detailed report. My plan is to fix this in version 3.3 (the next feature release). See https://lukemurphey.net/versions/320

**Comment by LukeMurphey** [comment]
This has been implemented in version 3.3.

You can now edit array fields:

![lookup_array](https://user-images.githubusercontent.com/1217581/57056802-cd6c2900-6c72-11e9-9fdb-de6b2bac8977.gif)

Note that you will need to make sure that no Javascript is cached in order to see that changes.
